gloss/Graphics/Gloss/Internals/Interface/Backend/GLFW.hs

```diff
@@ -481,36 +513,50 @@ callbackIdle stateRef callbacks
 
 
 -- Main Loop ------------------------------------------------------------------
-runMainLoopGLFW
-        :: IORef GLFWState
-        -> IO ()
 
-runMainLoopGLFW stateRef
- = X.catch go exit
- where
-  exit :: X.SomeException -> IO ()
-  exit e = print e >> exitGLFW stateRef
+runMainLoopGLFW :: IORef GLFWState -> IO ()
+runMainLoopGLFW stateRef = do
+        X.catch go handleException
+        GLFW.destroyWindow =<< windowHandle stateRef
+        GLFW.terminate
+
+    where
+        handleException :: X.SomeException -> IO ()
+        handleException = print
+
+        clearDirtyFlag :: IO ()
+        clearDirtyFlag = modifyIORef'
+                                stateRef
+                                (\state -> state { dirtyScreen = False })
+
+        whenDirty :: IO () -> IO ()
+        whenDirty action = do
+                dirty <- dirtyScreen <$> readIORef stateRef
+                when dirty $ do
+                        action
+                        GLFW.swapBuffers =<< windowHandle stateRef
```

Comment:
Hopefully it's not too on-the-nose to have `swapBuffers` called here. I thought the concept of "dirty" is pretty clearly going to result in a redraw in the Gloss context.